traits/Sluggable.php

```diff
@@ -0,0 +1,154 @@
+     *
+     * @param string $locale
+     *
+     * @return array
+     */
+    protected function translateSlugAttributes($locale)
+    {
+        $original = $this->getAttributes();
+
+        // Move translated values to model attributes
+        $this->translateContext($locale);
+        $this->fill($this->getTranslateAttributes($locale));
+
+        // Overwrite slug values with translated values in POST, if provided
+        foreach (array_keys($this->slugs) as $slugAttribute) {
+            if ($this->hasTranslation($slugAttribute, $locale)) {
+                $this->{$slugAttribute} = $this->getAttributeTranslated($slugAttribute, $locale);
+            } else {
+                $this->{$slugAttribute} = null;
+            }
+        }
+
+        // Generate slugs
+        $this->slugAttributes();
+
+        // Extract translated slugs and then return model to original state
+        $translatedSlugs = [];
+        foreach (array_keys($this->slugs) as $slugAttribute) {
+            $translatedSlugs[$slugAttribute] = $this->{$slugAttribute};
+        }
+
+        $this->translateContext(Translator::instance()->getDefaultLocale());
+        $this->fill($original);
+
+        return $translatedSlugs;
+    }
+
+    /**
+     * Ensures a unique attribute value, if the value is already used a counter suffix is added. Locale-aware.
+     *
+     * This will run the base Sluggable trait's method if the locale is the default locale.
+     *
+     * @param string $name The database column name.
+     * @param value $value The desired column value.
+     *
+     * @return string A safe value that is unique.
+     */
+    protected function getSluggableUniqueAttributeValue($name, $value)
+    {
+        if (!self::$currentLocale) {
+            return $this->baseGetSluggableUniqueAttributeValue($name, $value);
+        }
+
+        $counter = 1;
+        $separator = $this->getSluggableSeparator();
+        $_value = $value;
+
+        while (($this->methodExists('withTrashed') && $this->allowTrashedSlugs) ?
+            $this->newSluggableQuery()->transWhere($name, $_value, self::$currentLocale)->withTrashed()->count() > 0 :
+            $this->newSluggableQuery()->transWhere($name, $_value, self::$currentLocale)->count() > 0
+        ) {
+            $counter++;
+            $_value = $value . $separator . $counter;
+        }
+
+        return $_value;
+    }
+
+    /**
+     * Returns a query that excludes the current record if it exists. Locale-aware.
+     *
+     * This will run the base Sluggable trait's method if the locale is the default locale.
+     *
+     * @return Builder
+     */
+    protected function newSluggableQuery()
+    {
+        if (!self::$currentLocale) {
+            return $this->baseNewSluggableQuery();
+        }
+
+        return $this->exists
+            ? $this->newQuery()->where($this->getTable() . '.' . $this->getKeyName(), '<>', $this->getKey())
+            : $this->newQuery();
+    }
```

Comment:
I would incorporate the $this->methodExists('withTrashed') && $this->allowTrashedSlugs test in here and remove it from the getSluggableUniqueAttributeValue() method

Reply:
@mjauvin The code for `withTrashed` is a copy and slight tweak from the original Sluggable trait as [seen here](https://github.com/octobercms/library/blob/master/src/Database/Traits/Sluggable.php#L97). I'd like to try and keep it as close as possible to the original trait in order to make it easy to maintain (if we have to, say, make a change to Sluggable that needs to be done on both).

The `getTable()` call is required because the `transWhere()` call in `getSluggableUniqueAttributeValue()` brings in the translation tables as a relation in the SQL query. Without `getTable()`, an SQL error is thrown saying that the `id` column is ambiguous, so providing the table name removes the ambiguity.